user/advanced-topics/guivm.md

```diff
@@ -0,0 +1,134 @@
+
+[![sys-gui-gpu](/attachment/posts/guivm-gpu.png)](/attachment/posts/guivm-gpu.png)
+
+In `dom0`, enable the formula for `sys-gui-gpu` with pillar data:
+
+```bash
+sudo qubesctl top.enable qvm.sys-gui-gpu
+sudo qubesctl top.enable qvm.sys-gui-gpu pillar=True
+```
+
+then, execute it:
+
+```bash
+sudo qubesctl --all state.highstate
+```
+
+You can now disable the `sys-gui-gpu` formula:
+
+```bash
+sudo qubesctl top.disable qvm.sys-gui-gpu
+```
+
+At this point, you need to reboot your Qubes OS machine in order to boot into `sys-gui-gpu`.
+
+> None: For some platforms, it can be sufficient to shutdown all the running qubes and starting `sys-gui-gpu`. Unfortunately, it has been observed that detaching and attaching some GPU cards from `dom0` to `sys-gui-gpu` can freeze computer. We encourage reboot to prevent any data loss.
+
+Once, `lightdm` is started, you can log as `user` where `user` refers to the first `dom0` user in `qubes` group and with corresponding `dom0` password.
+
+### VNC GuiVM `sys-gui-vnc`
+
+Here, we describe how to setup `sys-gui-vnc` that we call a *remote* `GuiVM` or referenced as *with a virtual server* in [GUI Domain](https://www.qubes-os.org/news/2020/03/18/gui-domain/#virtual-server-the-perfect-remote-solution).
+
+[![sys-gui-vnc](/attachment/posts/guivm-vnc.png)](/attachment/posts/guivm-vnc.png)
+
+In `dom0`, enable the formula for `sys-gui-vnc` with pillar data:
+
+```bash
+sudo qubesctl top.enable qvm.sys-gui-vnc
+sudo qubesctl top.enable qvm.sys-gui-vnc pillar=True
+```
+
+then, execute it:
+
+```bash
+sudo qubesctl --all state.highstate
+```
+
+You can now disable the `sys-gui-vnc` formula:
+
+```bash
+sudo qubesctl top.disable qvm.sys-gui-vnc
+```
+
+At this point, you need to shutdown all your running qubes as the `default_guivm` qubes global property has been set to `sys-gui-vnc`. Then, you can start `sys-gui-vnc`:
+
+```bash
+qvm-start sys-gui-vnc
+```
+
+A VNC server session is running on `localhost:5900` in `sys-gui-vnc`. In order to reach the `VNC` server, we encourage to not connect `sys-gui-vnc` to a `NetVM` but rather to use another qube for remote access, say `sys-remote`. First, you need to bind port 5900 of `sys-gui-vnc` into a `sys-remote` local port (you may want to use another port than 5900 to reach `sys-remote` from the outside). For that, use `qubes.ConnectTCP` RPC service (see [Firewall](/doc/firewall). Then, you can use any `VNC` client to connect to you `sys-remote` on the chosen local port (5900 if you kept the default one). For the first connection, you will reach `lightdm` for which you can log as `user` where `user` refers to the first `dom0` user in `qubes` group and with corresponding `dom0` password.
+
+> Note: `lightdm` session remains logged even if you disconnect your `VNC` client. Ensure to lock or log out before disconnecting your `VNC` client session.
+
+### Troobleshooting
```

Comment:
I'm not sure where to put it, but https://github.com/QubesOS/qubes-issues/issues/5804 should be linked as a step required on the first use (to be fixed at some point...).

And generally, looking for [issues with `C: gui-domain` label](https://github.com/QubesOS/qubes-issues/issues?q=is%3Aopen+is%3Aissue+label%3A%22C%3A+gui-domain%22) is a good idea in case of troubles.

Reply:
Yes, I've it on my TODO list to possibly add `--all` options to `qvm-appmenus` to ease creation/update notably after switching to GuiVM.